apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -100,32 +108,192 @@ export class ChangeAction extends BaseRushAction {
         'determine which projects were changed. If this parameter is not specified, the checked out branch ' +
         'is compared against the "master" branch.'
     });
+
+    this._overwriteFlagParemter = this.defineFlagParameter({
+      parameterLongName: '--overwrite',
+      description: `If a changefile already exists, overwrite without prompting ` +
+        `(or erroring in ${BULK_LONG_NAME} mode).`
```

Comment:
My understanding is that `rush change` tracks multiple changes in a given PR.  For a given project, if you:
1. Do some work and commit
2. Run `rush change` and commit
3. Do some more work and commit
4. Run `rush change` again
...then you end up with two change files for that project, which seems like the right design.

However in certain cases (that I never understood) `rush change` overwrites an existing file rather than adding a new one.  Is this only if the file hasn't been committed yet?

We should clarify the behavior, and then make sure your `--overwrite` docs are consistent with that.

Reply:
Thanks @octogonz for pointing this out. I looked into the overwrite scenario you mentioned above - and it did feel confusing:
1) The changefile's filename follows the pattern `@package\name\branch_yyyy-mm-dd-hh-mm.json`. Now, if **rush change** is run twice in the same minute for the same package on the same branch, then the second run will generate the same changefile filename, leading to the "Overwrite?" prompt.
2) There is bug in the above flow! 😁 The file gets overwritten regardless of the option the user chooses at the "Overwrite?" prompt. The code looks like this:

```typescript
private _writeChangeFile() {
  if (FileSystem.exists(filePath)) {
    if (overwrite) {
      this._writeFile(filePath, output);
    } else {
      console.log(`Not overwriting ${filePath}...`);
    }
  }

  this._writeFile(filePath, output);
}
```

I will address this bug.

Reply:
Looks like the overwrite bug was already addressed in @iclanton's PR. There was another bug which I just fixed.